mne/viz/_brain/_brain.py

```diff
@@ -819,71 +819,39 @@ def set_data_smoothing(self, n_steps):
                     self._data[hemi]['smooth_mat'] = smooth_mat
         self.set_time_point(self._data['time_idx'])
 
-    @property
-    def _n_times(self):
-        return len(self._times) if self._times is not None else None
-
-    @property
-    def time_interpolation(self):
-        """The interpolation mode."""
-        return self._time_interpolation
-
-    @fill_doc
-    def set_time_interpolation(self, interpolation):
-        """Set the interpolation mode.
-
-        Parameters
-        ----------
-        %(brain_time_interpolation)s
-        """
-        _check_option('interpolation', interpolation,
-                      ('linear', 'nearest', 'zero', 'slinear', 'quadratic',
-                       'cubic'))
-        self._time_interpolation = str(interpolation)
-        del interpolation
-        self._time_interp_funcs = dict()
-        self._time_interp_inv = None
-        if self._times is not None:
-            idx = np.arange(self._n_times)
-            for hemi in ['lh', 'rh']:
-                hemi_data = self._data.get(hemi)
-                if hemi_data is not None:
-                    array = hemi_data['array']
-                    self._time_interp_funcs[hemi] = _safe_interp1d(
-                        idx, array, self._time_interpolation, axis=1,
-                        assume_sorted=True)
-            self._time_interp_inv = _safe_interp1d(idx, self._times)
-
-    def _interpolate_data(self, hemi, time_idx):
-        act_data = self._time_interp_funcs[hemi](time_idx)
-        act_time = self._time_interp_inv(time_idx)
+    def _interpolate_data(self, array, time_idx):
+        from scipy.interpolate import interp1d
+        times = np.arange(self._n_times)
+        act_data = interp1d(
+            times, array, self.interp_kind, axis=1,
+            assume_sorted=True)(time_idx)
+        ifunc = interp1d(times, self._data['time'])
+        act_time = ifunc(time_idx)
```

Comment:
This PR undid (again) some of the refactoring. The way my commits worked:

1. the interpolators were [set up once and reused](https://github.com/mne-tools/mne-python/pull/7257/commits/be72e644405bb5a0ece8b04fc8059730f76a22dc) (efficiency)
2. a helper [_safe_interp1d](https://github.com/mne-tools/mne-python/pull/7257/commits/da82289a1f031f7a28120caade5c5bb598b7897e) (avoids bugs now back in master)

And this commit appears to have un-done these. Accordingly we have lost the efficiency and the singleton regression is back in `master`:

https://circleci.com/gh/mne-tools/mne-python/18710?utm_campaign=workflow-failed&utm_medium=email&utm_source=notification


I think the real (only) bug that probably needed to be fixed here had to do with calling the interpolation functions incorrectly, right? If so @GuillaumeFavelier can you make a revert commit for this PR and try making the minimal changes required to get CIs happy again? I'd rather not sacrifice the `_safe_interp1d` and `funcs` efficiency gains...

Reply:
Ahh it's even more extensive -- I see this started out by completely reverting #7257. Are we planning on putting that back in somehow?

Reply:
I updated #7162 and delayed thoses features to 0.21 for now